src/Nethermind/Nethermind.DataMarketplace.Consumers/Shared/Services/ConsumerServicesBackgroundProcessor.cs

```diff
@@ -186,6 +182,18 @@ private async Task ProcessBlockAsync(long number, long timestamp)
             }
         }
 
+        private async Task<IEnumerable<DepositDetails>> GetDepositsToRefundAsync()
+        {
+            PagedResult<DepositDetails> depositsToRefundResult = await _depositManager.BrowseAsync(new GetDeposits
+            {
+                EligibleToRefund = true,
+                CurrentBlockTimestamp = _currentBlockTimestamp,
+                Results = int.MaxValue
```

Comment:
will this query grow unbounded and will return more and more deposits - all of them marked as eligible for refund
maybe they should not be marked as eligible for refund in the first place?